### ISSUE TITLE
Request design intent Check-in for Yellow Ribbon Migration

### ISSUE THREAD

**Issue by andaleliz** [department-of-veterans-affairs/va.gov-team#4470]
## Product Name

Doesn't really have a name that I'm aware of... Yellow Ribbon School Explorer is what I'm calling it in my mind :)
## Checkpoint Stage

[x] **Design Intent** (for early discussions about needs, priorities, concepts, research planning, and direction)

[ ] **Pre-usability testing Review** (for prototype and testing validation of best practices, alignment with va design system, and inclusive research best practices)

[ ] **Design Review** (for design work validation of best practices and alignment with the va design system)

### Where in the process is this? 

[x] Initial concept

[x] Initial mockup

[ ] Final mockup

## Design links or mockups

https://vsateams.invisionapp.com/freehand/document/ignBy6GSx
https://vsateams.invisionapp.com/share/K7VEJ2UE365
 
## Point of contact

Liz Lantz
## Preferred times

- Jan 2 between 1 and 4
- Jan 3 between 10:30-11:30, or after 1:30pm
- Jan 6 before 11:30 or after 2pm
- Jan 4 anytime before 3:30

## Next steps

This issue *automagically* creates a Zenhub ticket (assigned to Emily, Crystabel, and Jennifer (VSA Accessibility Specialist), and VSP will schedule a design review.

**Comment by jenstrickland** [comment]
**_VSA Accessibility Notes_**

_Mobile_
- LOVE THAT YOU STARTED WITH MOBILE! THANK YOU, FROM THE USERS OF AT & MOBILE!

- How did the card presentation come about? Was a table considered? It sounds like the data could be tabular, which adds semantic context for the users. For those who may be using translation software having that semantic info is also helpful, as it is for the most active non-sighted user of the web (Google/search engines).

- Comparison tool option; drawer or modal? There are some difficulties with the drawer functionality and screen readers that DK has referenced. Modals have similar issues. Would a separate page be possible?

_Desktop version_ 
— Was a table considered for this? Sorting could be done by table header. However, is that discoverable? Is the Sorting dropdown more intuitive for all? In another project a team found that the sorting dropdown was used by more participants than the tapping of column headers, but may be worth testing.

- The amount of space between elements on the card may be an issue for those with cognitive or low-vision issues. When using a magnification tool, they may not see the information on the right side. If you use a paper towel roll, close one eye, and look through it. That's how limited some users are. A table provides an affordance of more info.

_Questions_
- Is there an opportunity to move forward with MVP, but also conduct some user research to better understand the audience? Naturally, coming from accessibility, would love to prioritize those with disabilities to understand their needs, since they may be looking for schools that have certain parameters such as online/virtual learning, or other things.

- Instead of this tool going away, it may be an opportunity to explore what the needs of the audience actually are, to test hypotheses, and what tool is most helpful.

**Comment by andaleliz** [comment]
@jenstrickland Thanks for your thoughtful reply! Answers below:

> How did the card presentation come about? Was a table considered? 

(this is also in response to your question on tables for desktop) Card presentation came about from a desire align more with the GIBCT as I have a hypothesis that users will have seen that, and to consolidate the information,  into what I _think_ is more digestible. The repeated school names in the existing tables seems very, well, repetitive and tedious to go through.  Also, I'm inclined to avoid horizontal scrolling on mobile where possible, and have yet to find a solid responsive/accessible table solution where the tables collapse into "cards".  I also feel that tables are best suited for side-by-side comparisons, and I have another hypothesis that Veterans are in "search" mode in the first step, vs compare.

I felt that a table, once schools were selected to compare, was appropriate for that part of their research as they are truly indicating a desire to compare.

> Comparison tool, drawer or modal? Would a separate page be possible?

I was thinking a drawer, and sure, a separate page could be possible.

> Is there an opportunity to move forward with MVP, but also conduct some user research to better understand the audience?

Absolutely.  An updated data source will not be available to us until likely February, so we are going to test the UX on staging w/ the data we have in the interim to learn more about behaviors, needs, and how our proposed UI serves our users.

**Comment by CrystabelReiter** [comment]
I think this is going in the right direction.

I really like the solution for the mobile compare tool and I'm interested to see how it tests

I agree with you about the desktop version, I think the bottom drawer (or separate page) would be better than the mobile compare tool pattern on the lefthand side.

**Comment by andaleliz** [comment]
@CrystabelReiter Thanks! Glad we're on the same page.

@jenstrickland Do you have any guidance on how to frame participant criteria to those with disabilities are represented?  `At least one participant who uses assistive technology` is one I thought of.  Is there a disability rating target or anything like that?

**Comment by jenstrickland** [comment]
One more thing that occurred to me. In GIBCT, if I recall correctly, the headings are purple or dark blue. If these are links that is okay. If they are not links, they should not be dark blue or purple, as they will cognitively register as links. Since you're in sketch mode, this may not be relevant, and I'm noting only for future thought.

For research plan participants:
* Disability rating of 70% or more
* Participant who uses assistive technology (AT), such as screen readers, switches, translation, or magnification software (list of sample AT: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/accessibility/examples-of-assistive-tech.md)
* A mix of enlisted and officer (for socio-economic diversity), and geographic assortment

The inclusion of those who use AT is dependent on the format of your research and/or test prototype.

In a different review on Monday we saw what I think was an adjacent project that listed school names. The data that came in was only available in ALL CAPS. Do you know if the data you'll be working with has a similar challenge? Here's hoping that updated data source doesn't have ALL CAPS, since that incurs challenges for those with dyslexia, screenreader users, and everyone else. 

Regarding the table format. if a table is the right format for the purpose, let's not think about what can't be done, yet. It is possible to use the `<dl>` to render content to look like a table, and there may be other techniques we can use with a `<table>`. It may be worth testing card versus table layout, if time allows. The table would provide semantic structure to the data and enhance scan-ability.

**Comment by andaleliz** [comment]
Those suggestions on participants are exactly what I was looking for, thank you!

Our data does not have the ALL CAPS problem, thankfully.

I hear what you're saying on the tables, but I don't _think_ a table is the right format, until the user is side-by-side comparing schools, but that would be an excellent thing to test the two. At this point, I'm not sure what the timeline looks like once hi-fis are approved, so am not sure if we can develop two versions.

**Comment by 1Copenut** [comment]
@andaleliz I second the mobile-first approach! 

The things I'll focus on are related to micro-interactions and a possible pattern to explore:

* Love that you're using a Search button to drive the data call that returns results. This is a pattern we're trying to implement with GIBCT as well. Users get to drive the interaction and ask for results when they are ready. It'll cut down on unnecessary data transfers as well -- useful when paying by the kilobyte.
* I think it would be good to annotate or document where focus is set when data is returned and cards are populated. I think a good place to set focus is the `Displaying 1-10 of N results` text block. This will be invisible for mobile / desktop users, but will announce the data being returned to screen readers. In this case, I feel we can omit the yellow focus halo. @jenstrickland and I have been discussing this, so if it could be A/B tested with users, that'd be outstanding.
* What do you think about a focus-only "skip link" that jumps to the fixed bottom tray, right after the `Add to compare` button? I've used this pattern before, and it tested well with users. Basically we have a link that's hidden off left, and only appears on the page when it receives keyboard focus, like the top of page skip link. This would allow keyboard and screen reader users to skip to to the `Your Schools (2)` text block and open/close the tray quickly, without having to TAB through all the stops.
* If users press the Page 2 or Page 3 link, are you envisioning a full-page refresh, or just showing the second set of cards? If it's a partial page swap in React, you'll want to annotate where focus should go after data is loaded. I think it could be set back to `Displaying 11-20 of N results` block to announce the change and orient keyboard/assistive technology users.
* The pagination flow could also use a data loading strategy. If there are a large number of results, how big is the payload? Would it be worth loading the first 20 results on initial page load, so page 1 and page 2 could be pre-cached, and load page 3 if users click on page 2? I'm thinking this could be a hedge against slow first render / first interactivity.
* The Sort By menu should have a "Go" button or clear instructions inside it that user filters will be applied on click / check / radio select. Either is a good strategy to meet [SC 3.2.2 - On Input](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html#consistent-behavior-unpredictable-change-techniques-head)

**Comment by jenstrickland** [comment]
Re-opened issue, set assignees to designer, PM, and myself, to use as a working reference. Once @andaleliz   and @Karak888   feel they have reviewed feedback, they may close this issue, with a comment regarding decisions made, CC each of the reviewers in the comment for visibility.

**Comment by andaleliz** [comment]
@peggygannon , @mnorthuis do you have any feedback to add? I think you said you'd have some comments on content? Totally fine if that will come later, I believe there are a couple other reviews further in the process.

**Comment by andaleliz** [comment]
@1Copenut I could use some clarification on one of your comments. Thanks for all the feedback, btw.  

> What do you think about a focus-only "skip link" that jumps to the fixed bottom tray, right after the Add to compare button? I've used this pattern before, and it tested well with users. Basically we have a link that's hidden off left, and only appears on the page when it receives keyboard focus, like the top of page skip link. This would allow keyboard and screen reader users to skip to to the Your Schools (2) text block and open/close the tray quickly, without having to TAB through all the stops.

The way I'm understanding this, the skip link would be in each card. Is that correct? I really like this concept.

**Comment by peggygannon** [comment]
@andaleliz HI, Liz. Thank you for your presentation the other day. I've noted just two high level content recommendations at this time. I'll be providing more detailed feedback once you have a final mockup. Let me know if you have any questions. 


Current Title: Yellow Ribbon School Explorer. 
Here are some header suggestions you might want to also consider: 
--Yellow Ribbon School Finder
--Yellow Ribbon School Search
--Yellow Ribbon Program Tool

Also, in all references we should spell out Yellow Ribbon and not abbreviate to YR.

**Comment by 1Copenut** [comment]
@andaleliz Hey there. Here's follow ups to your questions. Holler at me if you want to chat or hop on a quick call.

* Yes, I'm proposing a skip link in each card. It's off-screen by default, and only appears when it receives keyboard focus. That way users who navigate with a mouse won't "find" it, but those who use the keyboard can skip down to the tray quickly.
* I might have mis-understood this second point. If you're sorting say the rows of a table by column ASC or DESC, that feels more like a single action that could be labeled with text and possibly SR-only text to explain what you are sorting by "Date", "Last Name" for instance.

Great questions!

**Comment by andaleliz** [comment]
@1Copenut The second point is sorting search results, not a table, but it's a similar concept where it's sorting ASC or DESC based on whichever facet they select (benefit amount, city, etc).

**Comment by jenstrickland** [comment]
@andaleliz @1Copenut I'm still wondering if a table might be a more accessible, semantic delivery of this information. It would be so much easier for screen reader and keyboard users to traverse the info, and others to visually scan data. We can style it to look more attractive, but benefit from all the aspects that come natively with a table. I can imagine way to present the info responsively, too.

**Comment by andaleliz** [comment]
Thanks to all of you for your input and suggestions! I’m delighted that our accessibility team is happy with the mobile-first approach and search button.  I've reviewed your feedback with both @Karak888 and @jenniferlee-dsva, and responses are below. We'll be conducting some user interviews and testing on this and, of course, will iterate on these decisions based on what we learn:
- We’ll stick with cards for search results, and and use a table for comparison. This is inline with other areas of va.gov.
- Agree that a page is a better solution than a drawer or modal, and the prototype has been updated.
- Focus annotation will be included in the hi-fi prototype, and we agree with recommendations about where to set focus, and removing the yellow halo when setting focus on the `Displaying 1-10 of N results` language for screen readers. 
- Agree with skip link in each card to ease keyboard navigation to comparison button
- Prototype updated with new suggested heading

**Comment by andaleliz** [comment]
@jenstrickland, I was composing my comment above as you sent yours. A table doesn't feel like the right presentation for search results to the team, but we will definitely present the comparison as a table. Our user testing may very well prove us wrong and we'll put the search results in a table if that's the case.

**Comment by jenstrickland** [comment]
@andaleliz If it's possible to test with keyboard and screen reader users to ensure it's still usable for them, that would be helpful. We have to move forward without doing as much research to understand the problem space and user scenarios as we'd like, so I get it!

**Comment by andaleliz** [comment]
@jenstrickland I've included that in my research plan. One thing that occurred to me though, is that I have never facilitated (or even observed) a study with users utilizing AT. Would you recommend that someone with experience in that arena facilitate those sessions, assuming we are able to find participants meeting that criteria for this study?

**Comment by jenstrickland** [comment]
@andaleliz I think we can reach out in vets-research channel for guidance. I'll contact you in slack for further discussion.

**Comment by mnorthuis** [comment]
@andaleliz sorry for the delay in response, just closing the loop on this (even though its already closed) that there is no IA feedback at this point in the process.
